### PR TITLE
Task 10 - added sort feature

### PR DESCRIPTION
**Pull request by dpeake99**

### CODE REVIEW

**Review by dappernerddesigns — APPROVED** [review]
You have a lot of repetitive code in your app now that it gets larger. You have hard coded links to topics and a separate component for viewing by topic which is a copy of articleList. You should be able to use articleList to render all articles, by topic and the sorted lists. By using the search params method outlined in the notes you would be able to refactor this into much less code.

Your sort state would then not be needed in App and could instead be declared in articleList and sent to the appropriate child component. You are currently left with prop drilling through files to get the state to your sorting drop downs.

While your sort _is _working__ this could be refactored to be more dry.

On mobile screens your app is overflowing and is not currently usable. Make sure to amend your css for smaller screens as you hit the later tickets.

File: src/utils/api.js [hunk 1]
```diff
@@ -3,8 +3,8 @@ import axios from "axios";
 const newsApi = axios.create({
     baseURL: "https://be-portfolio-project.herokuapp.com/api"
 })
-export const getArticles = () => {
-    return newsApi.get('/articles').then((res) => {
+export const getArticles = (sortedBy, orderedBy) => {
```
Comment (mallionaire) [inline src/utils/api.js:6]: You will be able to refactor this to use the axios params object which will append queries such as `sort)by` and `order` if they are defined. You will not need to provide these options in the request endpoint.

File: src/utils/api.js [hunk 2]
```diff
@@ -15,10 +15,14 @@ export const getTopics = () => {
     })
 }
 
-export const getArticlesByTopic = (topic) => {
-    return newsApi.get(`/articles?topic=${topic}`).then((res) => {
+export const getArticlesByTopic = (topic, sortedBy, orderedBy) => {
```
Comment (mallionaire) [inline src/utils/api.js:18]: If you use axios params it will remove the need fo rthis function entirely 😄

File: src/utils/api.js [hunk 3]
```diff
@@ -40,4 +44,10 @@ export const postNewComment = (articleId, comment, username) => {
     console.log("article=",articleId, "comment=", comment,"username=", username)
```
Comment (mallionaire) [inline src/utils/api.js:44]: no console.logs! Submitting a pull request is like saying "I think my code is production ready"

File: src/App.js [hunk 4]
```diff
@@ -2,19 +2,19 @@ import './App.css';
...
     <BrowserRouter>
         <div className="App">
           <Nav />
           <Routes>
-            <Route path ="/articles" element={<ArticleList/>}/>
-            <Route path="/cooking" element={<ArticleListByTopic topic={"cooking"}/>}/>
-            <Route path="/coding" element={<ArticleListByTopic topic={"coding"}/>}/>
-            <Route path="/football" element={<ArticleListByTopic topic={"football"}/>}/>
+            <Route path ="/articles" element={<ArticleList topic={""} />}/>
+            <Route path="/cooking" element={<ArticleList topic={"cooking"} />}/>
```
Comment (mallionaire) [inline src/App.js:15]: Routes for topics must not be hard-coded, please instead think of a way you can fetch the topics from your api and render a list (maybe mapping through the results!)

**Review by mallionaire — COMMENTED** [review]
There is still rather a lot of repetitive code here. You will need to refactor some of your api call to make use of the params object, there is a Q&A session that will cover this topic tomorrow 😄

**Pull request merged**